sharepoint-adfs/azuredeploy.json

```diff
@@ -0,0 +1,855 @@
+      }
+    },
+    "spSetupUserName": {
+      "type": "string",
+      "defaultValue": "spsetup",
+      "minLength": 1,
+      "metadata": {
+        "description": "Name of the service account that will create and configure the SharePoint farm"
+      }
+    },
+    "spSetupPassword": {
+      "type": "securestring",
+      "minLength": 8,
+      "metadata": {
+        "description": "Input must meet password complexity requirements as documented for property 'adminPassword' in https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/virtualmachines-create-or-update"
+      }
+    },
+    "spFarmUserName": {
+      "type": "string",
+      "defaultValue": "spfarm",
+      "minLength": 1,
+      "metadata": {
+        "description": "Name of the service account that will run SharePoint central administration and Timer services"
+      }
+    },
+    "spFarmPassword": {
+      "type": "securestring",
+      "minLength": 8,
+      "metadata": {
+        "description": "Input must meet password complexity requirements as documented for property 'adminPassword' in https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/virtualmachines-create-or-update"
+      }
+    },
+    "spSvcUserName": {
+      "type": "string",
+      "defaultValue": "spsvc",
+      "minLength": 1,
+      "metadata": {
+        "description": "Name of the service account that will run SharePoint service applications"
+      }
+    },
+    "spSvcPassword": {
+      "type": "securestring",
+      "minLength": 8,
+      "metadata": {
+        "description": "Input must meet password complexity requirements as documented for property 'adminPassword' in https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/virtualmachines-create-or-update"
+      }
+    },
+    "spAppPoolUserName": {
+      "type": "string",
+      "defaultValue": "spapppool",
+      "minLength": 1,
+      "metadata": {
+        "description": "Name of the service account that will run SharePoint web applications"
+      }
+    },
+    "spAppPoolPassword": {
+      "type": "securestring",
+      "minLength": 8,
+      "metadata": {
+        "description": "Input must meet password complexity requirements as documented for property 'adminPassword' in https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/virtualmachines-create-or-update"
+      }
+    },
+    "spPassphrase": {
+      "type": "securestring",
+      "minLength": 1,
+      "metadata": {
+        "description": "Passphrase of the SharePoint farm"
+      }
+    },
+    "vmDCName": {
+      "type": "string",
+      "defaultValue": "DC",
+      "minLength": 2,
+      "metadata": {
+        "description": "Internal name of DC VM"
+      }
+    },
+    "vmDCSize": {
+      "type": "string",
+      "defaultValue": "Standard_F4",
+      "metadata": {
+        "description": "Size of the DC VM"
```

Comment:
does the vm size need to account for premium storage options?  if so, probably good to mention in the desc